File: Suspension matlab/Suspension matlab raw/spring/Spring_stiffness.m
%% Spring stiffness vs natural frequency
clear; clc; close all;

%% Parameters

m = 250/4;   % quarter car mass (kg)

%% Spring stiffness range

k = linspace(10000,300000,1000); % N/m

%% Natural frequency

fn = (1/(2*pi)) * sqrt(k./m);

%% Plot

figure;
plot(k/1000,fn,'LineWidth',2);

grid on;
xlabel('Spring Stiffness (kN/m)');
ylabel('Natural Frequency (Hz)');
title('Natural Frequency vs Spring Stiffness');

File: Suspension matlab/spring/Spring_stiffness/Spring_stiffness.py
from pathlib import Path
import sys

import matplotlib.pyplot as plt
import numpy as np

ROOT = next(p for p in Path(__file__).resolve().parents if (p / "suspension_utils.py").exists())
sys.path.insert(0, str(ROOT))

from suspension_utils import load_data, show_plots


def main() -> None:
    p = load_data(Path(__file__).with_name("data.csv"))
    m = p["spring_stiffness_m"]
    k = np.asarray(p["spring_stiffness_k"], dtype=float)

    fn = (1 / (2 * np.pi)) * np.sqrt(k / m)

    plt.figure()
    plt.plot(k / 1000, fn, linewidth=2)
    plt.grid(True)
    plt.xlabel("Spring Stiffness (kN/m)")
    plt.ylabel("Natural Frequency (Hz)")
    plt.title("Natural Frequency vs Spring Stiffness")
    show_plots()


if __name__ == "__main__":
    main()
